app/controllers/api/v1/api_keys_controller.rb

```diff
@@ -47,19 +47,25 @@ def update
   private
 
   def check_mfa(user)
-    if user&.mfa_gem_signin_authorized?(otp)
+    if user&.webauthn_credentials&.any? && otp.blank?
```

Comment:
I'm a bit vague on the `&.` operator: Does `otp.blank?` still get tested if there are no `webauthn_credentials` on `user`?

Reply:
`&.` is the Safe Navigation Operator, meaning it will return `nil` instead of throwing an error if the user or webauthn credentials don't exist. Ruby utilizes a kind of short-circuiting for boolean expressions. So to answer your question, if there aren't any `webauthn_credentials`, or if the user returns nil, `otp.blank?` does not get tested. Does that answer your question?

Reply:
Got it. So if I am reading it correctly, this tests for whether the user has presented any kind of second factor (WebAuthn or TOTP)? If so there might be an opportunity to extract a variable, something like `user_did_not_present_second_factor`.

Reply:
Yep exactly, will extract to make things more readable.